src/test/java/com/amazon/ion/conformance/fragments.kt

```diff
@@ -264,12 +291,25 @@ private fun AnyElement.demangledWriteTo(writer: IonWriter) {
             writer.stepOut()
         }
         ElementType.SEXP -> {
-            if (sexpValues.firstOrNull().let { it is TextElement && it.textValue.startsWith("#$:") }) {
-                TODO("demangled e-expressions")
+            val head = sexpValues.firstOrNull()
+            if (head is TextElement && head.textValue.startsWith("#$:")) {
+                val tail = sexpValues.drop(1)
+                if (head.textValue == "#$::") {
+                    // Write an expression group
+                    writer as IonManagedWriter_1_1
+                    val rawWriter = writer.getRawUserWriter()
+                    rawWriter.stepInExpressionGroup(usingLengthPrefix = true)
```

Comment:
Do we need eventually need a way to exercise delimited expression groups here too...? This question is likely applicable in many places.

Reply:
I don't think so because this is a `toplevel` fragment, which is supposed to be encoding-agnostic. Expression groups should have full test coverage from other test cases that use `text` or `binary` fragments so that we can precisely control which text tokens or bytes are present in the encoding.